pyrealm/demography/crown.py

```diff
@@ -404,28 +347,85 @@ def calculate_stem_projected_leaf_area_at_z(
     return A_cp
 
 
-# def calculate_total_canopy_A_cp(z: float, f_g: float, community: Community) -> float:
-#     """Calculate total leaf area at a given height.
+@dataclass
+class CrownProfile:
+    """Calculate vertical crown profiles for stems.
 
-#     :param f_g:
-#     :param community:
-#     :param z: Height above ground.
-#     :return: Total leaf area in the canopy at a given height.
-#     """
-#     A_cp_for_individuals = calculate_projected_leaf_area_for_individuals(
-#         z, f_g, community
-#     )
+    This method calculates crown profile predictions, given an array of vertical
+    heights (``z``) for:
 
-#     A_cp_for_cohorts = A_cp_for_individuals * community.cohort_number_of_individuals
+    * relative crown radius,
+    * actual crown radius,
+    * projected crown area, and
+    * projected leaf area.
 
-#     return A_cp_for_cohorts.sum()
+    The predictions require a set of plant functional types (PFTs) but also the expected
+    allometric predictions of stem height, crown area and z_max for an actual stem of a
+    given size for each PFT.
 
+    Args:
+        stem_traits:
```

Comment:
I think this needs finished off

Reply:
This is what comes of trying to refactor the code in one large PR into five sequential sub-PRs on a Friday...

Reply:
Ah yes - I know what happened here. The Args docstring duplicates the attributes docstrings. I need to work out how to get the rendering so that only one of these needs to be provided but in the meantime I've duplicated.